### ISSUE TITLE
Icons missing in Windows high contrast model for FF

### ISSUE THREAD

**Issue by nicoxxie** [IBM/carbon-elements#345]
Received an a11y issue on the icon does not work in Firefox high contrast theme.  I verified this works on Chrome but not FF.

FF
![cdb81a52-2322-11e9-8352-9647c6936c8a](https://user-images.githubusercontent.com/17915105/52579473-95222f00-2e69-11e9-8dcb-8b6536663e64.png)

Chrome
![screen shot 2019-01-28 at 12 10 06 pm](https://user-images.githubusercontent.com/17915105/52575088-f265b280-2e60-11e9-879b-3b45661fe310.png)

https://github.ibm.com/IAM/access-management/issues/8027

**Comment by joshblack** [comment]
Hi @nicoxxie! 👋 Transferring this over to elements as this is where the new icons work will live 👍 

cc @dakahn do you know what causes this in high contrast mode? Is it the color of the icon itself doesn't have the right contrast ratio with the background? Or is it related to the icons themselves?

**Comment by icemanlily** [comment]
@joshblack @dakahn Any update for this issue?

**Comment by joshblack** [comment]
Could you all provide more details around how to reproduce this? e.g. what OS, which contrast mode / how to enable it, etc. Definitely still need to track this down!

cc @dakahn

**Comment by nicoxxie** [comment]
@joshblack Hi this was opened by QE, he posted the instructions in the ticket link I posted.

```
All important contents in a web page should be shown in Window High Contrast model.

How to apply a high-contrast theme in Windows
1> Press left Shift + left Alt + PrtSc (print screen) to open the High Contrast dialog.
2> Select the Go to Ease of Access Center link which opens the Make the computer easier to see page.
3> Select the Choose a High Contrast Theme link to open the Personalization control panel.
4> Select one of the high-contrast themes under Basic and High contrast themes. Recommendation: Use High Contrast #1 which is yellow on black for your test. For a more complete test, consider testing with more than one high contrast theme such as white on black.
5> Close the Personalization control panel.
6> Select Apply and OK buttons to apply the selection. Close the Ease of Access Center window, if still open.
```

**Comment by joshblack** [comment]
Thanks @nicoxxie! Will take a look, unfortunately nothing is coming to mind as to why this could be happening. Any thoughts on what it could be?

**Comment by dakahn** [comment]
I'm still looking into this but I'll leave some notes here so everyone's on the same page. Let's start with what Windows High Contrast mode is. It's _not_ inverted colors. It's what Microsoft describe as a "theme".

![2019-02-21 18_10_57-window](https://user-images.githubusercontent.com/40970507/53210617-19ac4480-3604-11e9-8851-39d33c47d1e4.png)

By default there are four color themes to choose from, but the user can tune each individual element's coloring (text, hyperlinks, disabled text etc) and apply those colors across their entire OS.

Firefox and Chrome have strikingly different behaviors when it comes to how they react toWindows High Contrast mode. 

Firefox detects the OS is in High Contrast Mode and overrides it's user interface and the page's styling with your chosen High Contrast Mode theme wholesale.

Chrome doesn't do this. If the user is in High Contrast Mode Chrome detects this and prompts the user to install a Google developed extension that leaves the page's base styling untouched and instead applies a "filter" to the page (much like a favorite extension of mine called Dark Reader). The user can select a few different filters to accommodate varying needs.

So what, right? Well I'm not sure yet. That icon is `rgb(21, 41, 53)` and with the Windows High Contrast Black theme enabled Firefox is converting that color to a very accessible `rgb(255, 255, 255)`. 

![2019-02-21 18_18_06-window](https://user-images.githubusercontent.com/40970507/53210964-3f861900-3605-11e9-90b1-327aab57dc49.png)

But it's also disappearing from the page. Upon further investigation it turns out that somehow the dev tools are lying and padlock is being rendered to something closer to it's original color. You can confirm this by changing the background color in a custom Windows High Contrast theme to a gray. You see this:

![2019-02-21 17_46_40-window](https://user-images.githubusercontent.com/40970507/53211144-dfdc3d80-3605-11e9-802f-21243e1132cc.png)

This seems like a bug on their end, but could be caused by something in our code. I will carry on looking into this and update with whatever I figure out.

**Comment by dakahn** [comment]
@nicoxxie @icemanlily 
Thanks for being cool and patient. For a fix you can target the SVG in a media query and check if Windows High Contrast Mode is active and then use a [CSS 2 system color](https://www.w3.org/TR/css-color-3/#css2-system) to access the high contrast theme. 

In this example we're grabbing the button text color and using the technique described in [this post ](https://css-tricks.com/accessible-svgs-high-contrast-mode/).

```
@media screen and (-ms-high-contrast: active) {
  .svg-icon {
    fill: ButtonText;
  }
}
```

We'll probably need to make this affordance within carbon-elements itself _somehow_ @joshblack, but I'm a little out of my depth for how to implement this change. Wanna pair up on a PR? 😺

**Comment by joshblack** [comment]
@dakahn just put up https://github.com/IBM/carbon-elements/pull/388 as an initial idea for what we could do. In theory, we could include it in elements and then components can inherit this style choice. Let me know what you think 👀

**Comment by joshblack** [comment]
Added in #388, work issue: https://github.com/IBM/carbon-components/issues/2104

Closing this in the meantime, will close 2104 when merged into stable!
